src/Test.UnitTests.Sarif/Core/RunTests.cs

```diff
@@ -487,9 +489,88 @@
~                }
~            }
~
             sb.Length.Should().Be(0, "failed test cases:\n" + sb.ToString());
         }
 
+        [Fact]
+        public void Run_ShouldSerializeAutomationDetails_WhenIdOrGuidIsNotNullOrWhiteSpace()
+        {
+            const string id = "automation-id";
+            const string guid = "automation-guid";
+
+            var sarifLog = new SarifLog
+            {
+                Runs = new[] { new Run() }
+            };
+            sarifLog.Runs[0].AutomationDetails = new RunAutomationDetails();
+
+            sarifLog.Runs[0].AutomationDetails.Id = id;
+            sarifLog.Runs[0].AutomationDetails.Guid = string.Empty;
+
+            AsserAutomationDetailsValues(sarifLog, id, string.Empty);
+
+            sarifLog.Runs[0].AutomationDetails.Id = string.Empty;
+            sarifLog.Runs[0].AutomationDetails.Guid = guid;
+
+            AsserAutomationDetailsValues(sarifLog, string.Empty, guid);
+
+            sarifLog.Runs[0].AutomationDetails.Id = id;
+            sarifLog.Runs[0].AutomationDetails.Guid = guid;
+
+            AsserAutomationDetailsValues(sarifLog, id, guid);
+        }
+
+        [Fact]
+        public void Run_ShouldNotSerializeAutomationDetails_WhenIdAndGuidAreNullOrWhiteSpace()
+        {
+            const string whiteSpace = " ";
+
+            var sarifLog = new SarifLog
+            {
+                Runs = new[] { new Run() }
+            };
+            sarifLog.Runs[0].AutomationDetails = new RunAutomationDetails();
+
+            sarifLog.Runs[0].AutomationDetails.Id = string.Empty;
+            sarifLog.Runs[0].AutomationDetails.Guid = string.Empty;
+
+            AsserAutomationDetailsValues(sarifLog, string.Empty, string.Empty);
+
+            sarifLog.Runs[0].AutomationDetails.Id = null;
+            sarifLog.Runs[0].AutomationDetails.Guid = null;
+
+            AsserAutomationDetailsValues(sarifLog, string.Empty, string.Empty);
+
+            sarifLog.Runs[0].AutomationDetails.Id = whiteSpace;
+            sarifLog.Runs[0].AutomationDetails.Guid = whiteSpace;
+
+            AsserAutomationDetailsValues(sarifLog, whiteSpace, whiteSpace);
```

Comment:
>AsserAutomationDetailsValues [](http://example.com/codeflow?start=21&length=28)

Your API is confusing. You have a log which already has an id and guid set on it. Why are you repeating those arguments instead of extracting them from the log?

If you just want the arguments, then instantiate the log file here. It's odd that you've split this work into two.

Reply:
thanks for the suggestion!
re-worked the method :)